DependencyInjection/HttplugExtension.php

```diff
@@ -467,4 +467,98 @@ public function getConfiguration(array $config, ContainerBuilder $container)
     {
         return new Configuration($container->getParameter('kernel.debug'));
     }
+
+    /**
+     * Configure a plugin using the parent definition from plugins.xml.
+     *
+     * @param ContainerBuilder $container
+     * @param string           $serviceId
+     * @param string           $pluginName
+     * @param array            $pluginConfig
+     *
+     * @return string configured service id
+     */
+    private function configurePlugin(ContainerBuilder $container, $serviceId, $pluginName, array $pluginConfig)
+    {
+        $pluginServiceId = $serviceId.'.plugin.'.$pluginName;
+
+        $definition = class_exists(ChildDefinition::class)
+            ? new ChildDefinition('httplug.plugin.'.$pluginName)
+            : new DefinitionDecorator('httplug.plugin.'.$pluginName);
+
+        $this->configurePluginByName($pluginName, $definition, $pluginConfig, $container, $pluginServiceId);
+        $container->setDefinition($pluginServiceId, $definition);
+
+        return $pluginServiceId;
+    }
+
+    /**
+     * Decorate the plugin service with a ProfilePlugin service.
+     *
+     * @param ContainerBuilder $container
+     * @param string           $pluginServiceId
+     */
+    private function decoratePluginWithProfilePlugin(ContainerBuilder $container, $pluginServiceId)
+    {
+        $container->register($pluginServiceId.'.debug', ProfilePlugin::class)
+            ->setDecoratedService($pluginServiceId)
+            ->setArguments([
+                new Reference($pluginServiceId.'.debug.inner'),
+                new Reference('httplug.collector.collector'),
+                new Reference('httplug.collector.formatter'),
+                $pluginServiceId,
+            ])
+            ->setPublic(false);
+    }
+
+    /**
+     * Configure a new StackPlugin for a client.
+     *
+     * @param ContainerBuilder $container
+     * @param string           $clientName
```

Comment:
this is used to configure the stack plugin so that we can show the right client name in the profiler, right? maybe mention this here...